### ISSUE TITLE
Coturn build compatibility with OpenSSL3 in FIPS mode

### ISSUE THREAD

**Issue by jb00ster** [coturn/coturn#836]
Hi, 
i'm trying to build the coturn with OpenSSL 3.0 & FIPS mode enabled

i got stuck on coturn build that fails with error : 
```
src/client/ns_turn_msg.c: In function 'stun_produce_integrity_key_str':
src/client/ns_turn_msg.c:260:7: warning: implicit declaration of function 'FIPS_mode' [-Wimplicit-function-declaration]
  260 |   if (FIPS_mode()) {
      |       ^~~~~~~~~
src/client/ns_turn_msg.c:260:7: warning: nested extern declaration of 'FIPS_mode' [-Wnested-externs]
...
/usr/bin/ld: lib/libturnclient.a(ns_turn_msg.o): in function `stun_produce_integrity_key_str':
/tmp/coturn-4.5.2/src/client/ns_turn_msg.c:260: undefined reference to `FIPS_mode' ⚠️ ⚠️ ⚠️ 
collect2: error: ld returned 1 exit status
make: *** [Makefile:121: bin/turnutils_oauth] Error 1
```

from file : 
https://github.com/coturn/coturn/blob/3865842d28d80ea3e61f5a247213755ad59e69d6/src/client/ns_turn_msg.c#L248-L250
and 
https://github.com/coturn/coturn/blob/3865842d28d80ea3e61f5a247213755ad59e69d6/src/client/ns_turn_msg.c#L260-L262

in fact, this function `FIPS_mode()` [was removed from OpenSSL 3](https://www.openssl.org/docs/man3.0/man7/migration_guide.html#Removed-FIPS_mode-and-FIPS_mode_set) and replaced with  `EVP_default_properties_is_fips_enabled()` 

for the moment i just replaced `FIPS_mode()` with `EVP_default_properties_is_fips_enabled(NULL)` 

this should be taken into account for futur release depending on OpenSSL version

thanks

**Comment by zjl158** [comment]
Hello, what is your final solution?

**Comment by eakraly** [comment]
This issue has been addressed in 4.6.0 (now fully supports openssl-3.0 and FIPS mode though not thoroughly tested)
